diff --git ext/bigdecimal/bigdecimal.c ext/bigdecimal/bigdecimal.c
index 7fdf40126e..ab89d5ee8d 100644
--- ext/bigdecimal/bigdecimal.c
+++ ext/bigdecimal/bigdecimal.c
@@ -103,54 +103,6 @@ static ID id_half;
 # define RB_OBJ_STRING(obj) StringValueCStr(obj)
 #endif

-#ifndef HAVE_RB_RATIONAL_NUM
-static inline VALUE
-rb_rational_num(VALUE rat)
-{
-#ifdef HAVE_TYPE_STRUCT_RRATIONAL
-    return RRATIONAL(rat)->num;
-#else
-    return rb_funcall(rat, rb_intern("numerator"), 0);
-#endif
-}
-#endif
-
-#ifndef HAVE_RB_RATIONAL_DEN
-static inline VALUE
-rb_rational_den(VALUE rat)
-{
-#ifdef HAVE_TYPE_STRUCT_RRATIONAL
-    return RRATIONAL(rat)->den;
-#else
-    return rb_funcall(rat, rb_intern("denominator"), 0);
-#endif
-}
-#endif
-
-#ifndef HAVE_RB_COMPLEX_REAL
-static inline VALUE
-rb_complex_real(VALUE cmp)
-{
-#ifdef HAVE_TYPE_STRUCT_RCOMPLEX
-  return RCOMPLEX(cmp)->real;
-#else
-  return rb_funcall(cmp, rb_intern("real"), 0);
-#endif
-}
-#endif
-
-#ifndef HAVE_RB_COMPLEX_IMAG
-static inline VALUE
-rb_complex_imag(VALUE cmp)
-{
-#ifdef HAVE_TYPE_STRUCT_RCOMPLEX
-  return RCOMPLEX(cmp)->imag;
-#else
-  return rb_funcall(cmp, rb_intern("imag"), 0);
-#endif
-}
-#endif
-
 #define BIGDECIMAL_POSITIVE_P(bd) ((bd)->sign > 0)
 #define BIGDECIMAL_NEGATIVE_P(bd) ((bd)->sign < 0)

diff --git ext/bigdecimal/bigdecimal.h ext/bigdecimal/bigdecimal.h
index 28f3363b3c..fdf112d769 100644
--- ext/bigdecimal/bigdecimal.h
+++ ext/bigdecimal/bigdecimal.h
@@ -92,14 +92,6 @@ llabs(LONG_LONG const x)
 }
 #endif

-#ifndef HAVE_FINITE
-static int
-finite(double)
-{
-    return !isnan(n) && !isinf(n);
-}
-#endif
-
 #ifndef isfinite
 # ifndef HAVE_ISFINITE
 #  define HAVE_ISFINITE 1
@@ -117,31 +109,6 @@ finite(double)
 # endif
 #endif

-#ifndef HAVE_RB_ARRAY_CONST_PTR
-static inline const VALUE *
-rb_array_const_ptr(VALUE a)
-{
-    return FIX_CONST_VALUE_PTR((RBASIC(a)->flags & RARRAY_EMBED_FLAG) ?
-	RARRAY(a)->as.ary : RARRAY(a)->as.heap.ptr);
-}
-#endif
-
-#ifndef RARRAY_CONST_PTR
-# define RARRAY_CONST_PTR(a) rb_array_const_ptr(a)
-#endif
-
-#ifndef RARRAY_AREF
-# define RARRAY_AREF(a, i) (RARRAY_CONST_PTR(a)[i])
-#endif
-
-#ifndef HAVE_RB_SYM2STR
-static inline VALUE
-rb_sym2str(VALUE sym)
-{
-    return rb_id2str(SYM2ID(sym));
-}
-#endif
-
 #ifndef ST2FIX
 # undef RB_ST2FIX
 # define RB_ST2FIX(h) LONG2FIX((long)(h))
diff --git include/ruby/missing.h include/ruby/missing.h
index f83f1b695c..200f365d37 100644
--- include/ruby/missing.h
+++ include/ruby/missing.h
@@ -96,10 +96,6 @@ RUBY_EXTERN int eaccess(const char*, int);
 RUBY_EXTERN double round(double);	/* numeric.c */
 #endif

-#ifndef HAVE_FINITE
-RUBY_EXTERN int finite(double);
-#endif
-
 #ifndef HAVE_FLOCK
 RUBY_EXTERN int flock(int, int);
 #endif
