diff --git include/fpu_control_x86.h include/fpu_control_x86.h
index c20e0ac..5967b83 100644
--- include/fpu_control_x86.h
+++ include/fpu_control_x86.h
@@ -16,6 +16,7 @@
    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
 
 #ifndef _FPU_CONTROL_H
 #define _FPU_CONTROL_H	1
