diff --git src/gui/menu_osx.mm src/gui/menu_osx.mm
index 985245b99..9a124bd58 100644
--- src/gui/menu_osx.mm
+++ src/gui/menu_osx.mm
@@ -523,7 +523,7 @@ int my_quartz_match_window_to_monitor(CGDirectDisplayID *new_id,NSWindow *wnd) {
             pt.y = (prct.origin.y + prct.size.height) - pt.y;
         }
 
-        err = CGGetDisplaysWithPoint(pt,1,&did,&cnt);
+        err = CGGetDisplaysWithPoint(CGPointMake(pt.x,pt.y),1,&did,&cnt);
 
         /* This might happen if our window is so far off the screen that the center point does not match any monitor */
         if (err != kCGErrorSuccess) {
