diff --git src/gui/menu_osx.mm src/gui/menu_osx.mm
index 985245b99..cbdb75ce3 100644
--- src/gui/menu_osx.mm
+++ src/gui/menu_osx.mm
@@ -515,7 +515,16 @@ int my_quartz_match_window_to_monitor(CGDirectDisplayID *new_id,NSWindow *wnd) {
 // NTS: This did not appear until Mojave, and some followers on Github prefer to compile for somewhat older versions of OS X
 //      NSPoint pt = [wnd convertPointToScreen:NSMakePoint(rct.size.width / 2, rct.size.height / 2)];
 // NTS: convertRectToScreen however is documented to exist since 10.7, unless Apple got that wrong too...
-        NSPoint pt = [wnd convertRectToScreen:NSMakeRect(rct.size.width / 2, rct.size.height / 2, 0, 0)].origin; /* x,y,w,h */
+        NSPoint pt = {0,0};
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
+        if ([wnd respondsToSelector:@selector(convertRectToScreen:)]) {
+             pt = [wnd convertRectToScreen:NSMakeRect(rct.size.width / 2, rct.size.height / 2, 0, 0)].origin; /* x,y,w,h */
+        } else {
+             pt = [wnd convertBaseToScreen:NSMakeRect(rct.size.width / 2, rct.size.height / 2, 0, 0).origin]; /* x,y,w,h */
+        }
+#else
+        pt = [wnd convertBaseToScreen:NSMakeRect(rct.size.width / 2, rct.size.height / 2, 0, 0).origin]; /* x,y,w,h */
+#endif
 
         {
             /* Eugh this ugliness wouldn't be necessary if we didn't have to fudge relative to primary display. */
